DESIGN.md

```diff
@@ -0,0 +1,108 @@
+
+Directory="<%= get_service("marionette-collective").bindir_id %>"
+
+More information about the actual specific WiX services can be found under the ServiceControl/Install/Config children of the wix element [component](http://wixtoolset.org/documentation/manual/v3/xsd/wix/component.html)
+
+### Directorylist.wxs
+In order to facilitate the bindir_id method in components, vanagon uses a WiX file and platform function to create a list of the service directories. The file (directorylist.wxs in vanagon) operates using non-named directory elements. Vanagon uses the fact that If you create a directory element with no name attribute, the element won’t exist as a new directory, but a reference to its parent element’s directory.
+
+The directorylist file itself is simple, it creates the necessary root directories in WiX, then inside the last directory it calls out to Vanagon::Platform::Windows.generate_service_bin_dirs. This function does a two things for each service file:
+
+1. Generates the necessary directories underneath where the file is, these directory elements will have names and resolve to a path
+2. Generates one last element underneath the top direcotory with no name and bindir_id as the id element. This provides the necessary connection between bindir_id and the actual directory containing binaries. As discussed before because this element has no name it will resolve to it’s parent directory element, the top level directory.
+
+In cases where there is more than one bindir_id corresponding to the same place there is no problem, there will just be two nameless directory elements under that directory. This is allowed in WiX because they have no names.
+
+### INSTALLDIR
+Inside the directorylist file the last root directory defined has the unique id INSTALLDIR. Vanagon uses this name on purpose. Used correctly, INSTALLDIR provides the user the means to change the location of the final install of the MSI. Using WiX properties and the INSTALLDIR variable the user can override all root directories underneath INSTALLDIR with what the end user wants.
+
+Note that this is in no way the correct way for the vanagon user to override directorylist root install defaults. If the user wishes the default installation location to be something other than DRIVE:/ProgramFilesFolder/CompanyName/ProductName they should override the entire directorylist file with a file of the same name in their own project. Vanagon will default to using the one in their project.
+
+INSTALLDIR itself is not a WiX property or requirement, the name was a convention that vanagon uses.
```

Comment:
Put in a link/reference to this.
Its not a standard MSI property, but has become a kind of de-facto standard after Installshield brought it into common usage.